pyccl/covariances.py

```diff
@@ -127,3 +129,240 @@ def angular_cl_cov_cNG(cosmo, cltracer1, cltracer2, ell, tkka, fsky=1.,
+            `None`, a default sampling will be used.
+        fsky (float): sky fraction.
+        p_of_k_a (:class:`~pyccl.pk2d.Pk2D`, str, or `None`): Linear
+            power spectrum to use. Defaults to `None`.
+
+    Returns:
+        float or array_like: values of the projected variance.
+    """
+    status = 0
+    if a is None:
+        na = lib.get_pk_spline_na(cosmo.cosmo)
+        a_arr, status = lib.get_pk_spline_a(cosmo.cosmo, na, status)
+        check(status)
+    else:
+        a_arr = np.atleast_1d(a)
+        na = len(a_arr)
+
+    chi_arr = comoving_radial_distance(cosmo, a_arr)
+    R_arr = chi_arr * np.arccos(1-2*fsky)
+    psp = parse_pk2d(cosmo, p_of_k_a, is_linear=True)
+
+    s2B_arr, status = lib.sigma2b_vec(cosmo.cosmo, a_arr, R_arr, psp,
+                                      na, status)
+    check(status, cosmo=cosmo)
+    if a is None:
+        return a_arr, s2B_arr
+    else:
+        if np.ndim(a) == 0:
+            return s2B_arr[0]
+        else:
+            return s2B_arr
+
+
+def sigma2_B_from_mask(cosmo, a=None, mask_wl=None, p_of_k_a=None):
+    """ Returns the variance of the projected linear density field, given the
+        angular power spectrum of the footprint mask and scale factor.
+        This is given by
+
+    .. math::
+        \\sigma^2_B(z) = \\frac{1}{\\chi^2{z}}\\sum_\\ell
+            P_L(\\frac{\\ell+\\frac{1}{2}}{\\chi(z)},z)\\,
+            (2\\ell+1)\\sum_m W^A_{\\ell m} {W^B}^*_{\\ell m},
+
+    where :math:`W^A_{\\ell m}` and :math:`W^B_{\\ell m}` are the spherical
+    harmonics decomposition of the footprint masks of fields `A` and `B`,
+    normalized by their area.
+
+    Args:
+        cosmo (:class:`~pyccl.core.Cosmology`): a Cosmology object.
+        a (float, array_like or `None`): an array of scale factor
+            values at which to evaluate the projected variance.
+        mask_wl (array_like): Array with the angular power spectrum of the
+            masks. The power spectrum should be given at integer multipoles,
+            starting at :math:`\\ell=0`. The power spectrum is normalized
+            as :math:`(2\\ell+1)\\sum_m W^A_{\\ell m} {W^B}^*_{\\ell m}`.
+        p_of_k_a (:class:`~pyccl.pk2d.Pk2D`, str, or `None`): Linear
+            power spectrum to use. Defaults to `None`.
+
+    Returns:
+        float or array_like: values of the projected variance.
+    """
+    if p_of_k_a is None:
+        cosmo.compute_linear_power()
+        p_of_k_a = cosmo.get_linear_power()
+
+    a_arr = np.atleast_1d(a)
+    chi = comoving_angular_distance(cosmo, a=a_arr)
+
+    ell = np.arange(mask_wl.size)
+
+    sigma2_B = np.zeros(a_arr.size)
+    for i in range(sigma2_B.size):
+        if 1-a_arr[i] < 1e-6:
+            # For a=1, the integral becomes independent of the footprint in
+            # the flat-sky approximation. So we are just using the method
+            # for the disc geometry here
+            sigma2_B[i] = sigma2_B_disc(cosmo=cosmo, a=a_arr[i],
```

Comment:
Humm, shouldn't this depend on fsky somehow?

Reply:
This only kicks in for chi=0, for which Rs = 0, and kR=0. The tophat window is 1 in this case, independent of fsky.

Reply:
oh, I see, thanks